lib/World.luau

```diff
@@ -210,28 +340,28 @@ end
 
 function World:_transitionArchetype(id, components)
 	debug.profilebegin("transitionArchetype")
+	local storage = self.storage
 	local newArchetype = nil
 	local oldArchetype = self._entityArchetypes[id]
-	local oldStorage
 
 	if oldArchetype then
-		oldStorage = self:_getStorageWithEntity(oldArchetype, id)
-
 		if not components then
-			oldStorage[oldArchetype][id] = nil
+			storage[oldArchetype][id] = nil
 		end
 	end
 
 	if components then
 		newArchetype = archetypeOf(unpack(self._entityMetatablesCache[id]))
 
 		if oldArchetype ~= newArchetype then
-			if oldStorage then
-				oldStorage[oldArchetype][id] = nil
+			-- NOTE:
+			-- This seems... maybe wrong
+			if oldArchetype then
+				storage[oldArchetype][id] = nil
```

Comment:
After looking into it, I'm pretty sure this is correct. We must remove the archetype from where it belongs right now -- this previously was always in the old storage because that's how the strategy worked.

Reply:
The old storage no longer exists. There is no need to move the archetype from old to pristine storage, only to transition the archetype to the new one.

Reply:
This is required to transition the archetype to the new one. We have to remove the entity from the old archetype.